src/filebrowser/PasMetadata.vue

```diff
@@ -0,0 +1,257 @@
+							<option :value="undefined" disabled>-- Select an encoding --</option>
+						</template>
+					</b-form-select>
+				</b-form-group>
+			</b-col>
+
+			<b-col lg="6">
+				<b-form-group label="Sequential settings:">
+					<b-form-checkbox class="" v-model="sequential" name="check-button" switch>
+						This file is a sequential file (for example csv file)
+					</b-form-checkbox>
+				</b-form-group>
+			</b-col>
+		</b-row>
+
+		<b-row v-if="sequential" style="margin-bottom: 10px;">
+			<b-col lg="6">
+				<b-form-group label="Delimiter:">
+					<b-form-select  v-model="edits.delimiter" :options="delimiterOptions">
+						<template slot="first">
+							<option :value="undefined" disabled>-- Select a delimiter --</option>
+						</template>
+					</b-form-select>
+				</b-form-group>
+			</b-col>
+			<b-col lg="6">
+				<b-form-group label="Quoting character (defaults: \ ):" >
+					<b-form-input v-model="edits.quotingCharacter" placeholder="Enter quoting character"></b-form-input>
+				</b-form-group>
+			</b-col>
+		</b-row>
+
+		<b-row v-if="sequential" style="margin-bottom: 10px;">
+			<b-col lg="6">
+				<b-form-group label="Record separator:">
+					<b-form-select v-model="edits.separator" :options="separatorOptions">
+						<template slot="first">
+							<option :value="undefined" disabled>-- Select a record separator --</option>
+						</template>
+					</b-form-select>
+				</b-form-group>
+			</b-col>
+			<b-col lg="6">
+				<b-form-group label="Header:">
+					<b-form-checkbox v-model="edits.header" name="check-button" switch>
+						Header in file?
+					</b-form-checkbox>
+				</b-form-group>
+			</b-col>
+
+		</b-row>
+
+		<p class="info-text" v-if="file"> Note! By saving the data you are changing the file metadata regardless whether you select the file into your dataset or not. Dataset is not saved / published!</p>
+		<p class="info-text" v-else> Note! By saving the data you are changing the file metadata FOR ALL THE FILES IN THE FOLDER regardless whether you select the files / folder into your dataset or not. Dataset is not saved / published!</p>
+		<b-alert :show="saveResult === 'failed'" variant="danger" dismissible>Something went wrong. Your changes are not saved. Please try again.</b-alert>
+		<b-alert :show="saveResult === 'success'" variant="success" dismissible>The file metadata was successfully saved.</b-alert>
+		<b-button @click="save" variant="primary" class="w-100 h-100">Save PAS Metadata</b-button>
+	</b-card>
+</template>
+
+<style lang="scss" scoped>
+
+.first-row {
+	padding: 5px;
+	> * {
+		display: inline-block;
+		width: 30%;
+	}
+	> *:not(:first-child) {
+		margin-left: 5px;
+	}
+}
+.info-text {
+	padding-bottom: 20px;
+	padding-top: 20px;
+	padding-left: 60px;
+	padding-right: 60px;
+}
+</style>
+
+<style lang="scss">
+table#file-table thead th {
+    border-top: 0px;
+}
+</style>
+
+
+
+<script>
+import axios from 'axios'
+import dateFormat from 'date-fns/format'
+
+const fileAPI = axios.create({
```

Comment:
we create a new instance here, where do we destroy it? What is the lifecycle of this instance?

Reply:
As far as I know, it dies with the component instance.

Reply:
for my eyes there is a risk that we create a new instance into non component javascript scope.

Also after reading the API documents I wonder what is the technical reason why we need this custom instance. I presume that the original purpose was to set some global configuration which applies to all requests.

For that purpose there is following pattern:
```
axios.defaults.baseURL = process.env.VUE_APP_METAX_FILEAPI_URL;
axios.defaults.timeout = 3000;
axios.defaults.responseType = 'json';
```
https://github.com/axios/axios#global-axios-defaults

This would then allow us to use this library without a custom instance as well.

Reply:
I have also seen that there is this:
https://github.com/CSCfi/qvain-js/blob/next/src/api/client.js

which is used elsewhere, maybe that would be a better pattern here.
Then this client communication would be defined in a single place instead.

That is used for example in
```
./components/VersionsModal.vue:26:import apiClient from '@/api/client.js'
./components/VersionsModal.vue:66:			apiClient.get(`/datasets/${dataset}/versions`)
```